### PR TITLE
Improve docs and docstring accuracy

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

Systematic accuracy pass over user-facing text: docs pages, Python docstrings, inline comments, and example code. 90 small, high-confidence corrections — no behavior changes, no restyling, no restructuring. Every fix was verified against the authoritative source before inclusion: actual code paths and signatures, `ultralytics/cfg/default.yaml`, dataset YAMLs, the YOLOv5 repository for legacy docs, and the Platform app for platform docs. Ambiguous or lower-confidence findings were deliberately left out to keep this reviewable.

## Correction categories

- **Broken examples**: MobileSAM box-prompt section used point prompts; SAM predictor snippet missing `import cv2`; `torch2coreml()` call missing required `classifier_names`; `YOLOEDetect`/`YOLOESegment` docstring examples passed text embeddings as a second positional arg; `prompt_inference` example unpacked 3 returns (returns 2); `Results.plot()` example called `.show()` on an ndarray; OBB tracking example used `tracker=True` on predict instead of `model.track()`; `check_cfg` docstring example raised instead of passing
- **Docs vs code mismatches**: `benchmark()` device/imgsz docs; DEEPX calibration default `coco8.yaml` and unsupported `fraction` advice; ExecuTorch FAQ contradicted its own Predict/Validate examples; settings table (`settings_version` 0.0.6, `tensorboard`/`wandb` default `False`); `batch` vs nonexistent `batch_size`; solutions CLI list missing `region`/`security`/`parking`; solution outputs save to `runs/solutions/exp`; analytics output 1280×720; detection fitness weights; `ObjectCropper` filenames; `make_divisible` rounds up; `process_mask` output resolution; pose val keypoints scale to model-input space; OBB loss comment `xywhr (b, h*w, 5)`; autobatch fit is linear, not log-space; `Colors` palette is not TABLEAU; `ConfusionMatrix` task values; `SolutionResults` image attr is `plot_im`
- **Format/coordinate precision**: `imgsz` lists are `[h, w]`; pose "3D keypoints" → keypoint visibility; normalized vs pixel keypoint coords; HSV comment BGR not RGB; `polygon2mask` flat 1D polygon input; semantic mask extension resolution; classify split roles (`train`/`val`, optional `test`)
- **Invalid CLI/flags**: `yolo export model=... format=...` key=value form; `yolo settings vscode_msg=False`; YOLOv5 docs advertised nonexistent `--amp` and `--close-mosaic` flags; YOLOv5 evolve output `runs/evolve/exp/hyp_evolve.yaml`; run dirs `exp2` not `exp-2`; CPU docker run without `--runtime=nvidia`; `hyp.VOC.yaml` not `hyp.finetune.yaml`
- **Wrong links/names**: FastSAM repo org (CASIA-IVA-Lab); `autosplit` moved to `reference/data/split`; race-conditions text linked to algorithmic-bias glossary; RT-DETR = Real-Time DEtection TRansformer; `no_goggle` class name; YOLOv8 Explore row missing OBB
- **Stale values**: YOLO26n-cls 71.4%/90.1%; SAHI slice defaults 512; examples retirement note; `README.zh-CN.md` pose mAP 71.6 and semantic reproduce command

## Changed areas

- `docs/en/**` — 36 files (models, datasets, guides, integrations, modes, platform, solutions, usage, quickstart, yolov5)
- `ultralytics/**` — 22 files (cfg, data, engine, models, nn, solutions, utils — docstrings/comments only; one return annotation corrected to match the documented return, no runtime changes)
- `examples/**` — 5 READMEs; `README.zh-CN.md`

## Validation

- `git diff --check` clean
- `python -m py_compile` passes on all 22 edited Python files
- Relative-link + code-fence integrity check across all 36 changed markdown files: 0 broken links, all fences balanced
- Platform docs verified against the Platform app; YOLOv5 docs verified against ultralytics/yolov5; vendor-dependent claims verified against current vendor docs

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📝 This PR is primarily a documentation and example cleanup update that improves accuracy, fixes outdated commands and references, and aligns docs with current Ultralytics behavior across YOLO26, datasets, exports, solutions, and utilities.

### 📊 Key Changes
- 📚 Updated many docs pages to correct outdated or misleading instructions, parameter names, defaults, and examples.
- 🎯 Refined dataset documentation:
  - Classification split structure now correctly emphasizes `train` + `val`, with `test` optional.
  - Pose format wording now describes visibility flags more clearly.
  - Semantic segmentation docs now better explain mask file handling and recommend lossless formats like `.png`/`.tiff`.
  - Minor dataset label fixes such as `no_goggle` in Construction-PPE.
- 🤖 Refreshed model and benchmark docs:
  - Updated YOLO26 classification accuracy numbers in ImageNet docs.
  - Corrected benchmark docs to state that `benchmark()` only supports square `imgsz`.
  - Fixed model YAML docs to reflect current `Detect` arguments and `torchvision.ops.` module resolution.
- 🛠️ Improved command and API examples across guides:
  - Fixed CLI syntax like `yolo export model=yolo26n.pt format=tflite` and `yolo settings vscode_msg=False`.
  - Replaced incorrect `batch_size` references with the actual `batch` argument.
  - Updated CoreML export example to include `classifier_names=None`.
  - Corrected autosplit import/reference paths.
- 🧠 Clarified training and validation behavior:
  - Custom trainer docs now describe the current detection fitness metric more accurately.
  - Several comments/docstrings now better explain pose scaling, mask processing, class weights, confusion matrix task types, and divisibility behavior.
- 📦 Updated integration and deployment docs:
  - DEEPX docs now use `coco8.yaml` as the calibration dataset default and simplify calibration wording.
  - ExecuTorch docs now state that exported models can be loaded with `YOLO()` in Python for inference/validation.
- ✂️ Refined SAM and MobileSAM examples:
  - MobileSAM examples now focus on box prompts instead of point prompts.
  - SAM examples were corrected for prompt label formatting and imports.
- 🧩 Expanded solutions documentation:
  - Added newer solution names like `region`, `security`, and `parking` to the documented CLI list.
  - Clarified solutions output paths and analytics output sizing in code comments/docstrings.
- 🧹 Cleaned up examples and legacy docs:
  - Updated example READMEs to better match current behavior, paths, defaults, and expected outputs.
  - Softened the deprecation wording for the `examples/` directory.
  - Fixed several YOLOv5 tutorial paths, filenames, and training tips.
- 🔍 Included many small consistency fixes:
  - README metric reproduction commands, model tables, links, filenames, comments, and typos were corrected throughout the repo.

### 🎯 Purpose & Impact
- ✅ Makes the documentation more trustworthy by aligning it with actual current code behavior and defaults.
- 🚀 Reduces user confusion and setup mistakes caused by outdated commands, wrong file paths, or inaccurate parameter names.
- 🧪 Helps users reproduce results more reliably with corrected benchmark, validation, export, and dataset instructions.
- 📖 Improves accessibility for both new and experienced users by making explanations clearer and examples more practical.
- 🛡️ Low technical risk: this PR appears to focus mostly on docs, examples, comments, and clarifications rather than changing runtime logic.
- 🌍 Improves overall polish across the Ultralytics ecosystem, especially for YOLO26, integrations, solutions, and educational guides.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an engineer will assist shortly. Please review the checklist below to help ensure a smooth merge. ✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Most of the PR is a solid pass over outdated docs/docstrings, but I found a few regressions worth fixing before merge: the FastSAM validation example now uses an incompatible multi-class dataset, the semantic-segmentation docs newly suggest lossy mask formats, one `check_cfg()` docstring example still doesn't match runtime coercion, and the C++ README now references the standalone YOLOv5 repo without updating the downstream path instructions.

💬 Posted 4 inline comments

File: docs/en/datasets/semantic/index.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Semantic segmentation datasets use one image file and one mask file per sample.
 - Pixel values `0`, `1`, `2`, ... represent class IDs from the dataset `names` mapping.
 - Pixel value `255` is treated as the ignore label and is excluded from loss and metric computation.
 - Mask files should use the same stem as their matching image file, for example `frankfurt_000000_000294.png`.
-- Supported mask extensions are `.png`, `.PNG`, `.bmp`, and `.tif`.
+- Masks are resolved as `.png` by default; if missing, any extension in the supported image formats (e.g. `.png`, `.jpg`, `.jpeg`, `.bmp`, `.tif`, `.tiff`, `.webp`) is also accepted.
```
Comment (UltralyticsAssistant) [inline docs/en/datasets/semantic/index.md]: 💡 **MEDIUM**: Including lossy formats like `.jpg`, `.jpeg`, and `.webp` here is risky for semantic masks because mask pixels encode class IDs exactly. Even if the loader can fall back to generic image extensions, documenting lossy formats as acceptable mask files can lead users to silently corrupt labels. I'd keep the docs restricted to lossless mask formats such as PNG/TIFF.

File: docs/en/models/fast-sam.md [hunk 2]
```diff
@@ -192,7 +192,7 @@ Validation of the model on a dataset can be done as follows:
 
         ```bash
         # Load a FastSAM model and validate it on the COCO8 example dataset at image size 640
-        yolo segment val model=FastSAM-s.pt data=coco8.yaml imgsz=640
+        yolo segment val model=FastSAM-s.pt data=coco8-seg.yaml imgsz=640
```
Comment (UltralyticsAssistant) [inline docs/en/models/fast-sam.md:195]: 💡 **MEDIUM**: This example now points users at `coco8-seg.yaml`, but the note immediately below says FastSAM only supports a single class and requires all category IDs to be remapped to `0`. `coco8-seg` is a multi-class segmentation dataset, so this command is likely to produce misleading validation results unless the labels are converted first. Please switch the example to a single-class seg dataset or call out the required remapping inline.

File: examples/YOLOv8-CPP-Inference/README.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ yolo export model=yolov8s.pt imgsz=640,480 format=onnx opset=12 # Example: 640x4
 
 **Exporting YOLOv5 Models:**
 
-Use the `export.py` script from the YOLOv5 repository structure (included within the cloned `ultralytics` repo).
+Use the `export.py` script from the separate [YOLOv5 repository](https://github.com/ultralytics/yolov5).
```
Comment (UltralyticsAssistant) [inline examples/YOLOv8-CPP-Inference/README.md:56]: 📝 **LOW**: Now that this points readers to the separate YOLOv5 repository, the follow-on command block/comment below is no longer self-consistent because it still assumes the `ultralytics` repo layout. Please update the surrounding instructions to use the YOLOv5 repo root as the working directory so the export steps remain copy-pasteable.

File: ultralytics/cfg/__init__.py [hunk 4]
```diff
@@ -363,12 +363,12 @@ def check_cfg(cfg: dict, hard: bool = True) -> None:
...
         ...     "epochs": 50,  # valid integer
         ...     "lr0": 0.01,  # valid float
-        ...     "momentum": 1.2,  # invalid float (out of 0.0-1.0 range)
+        ...     "momentum": 0.937,  # valid float
         ...     "save": "true",  # invalid bool
         ... }
         >>> check_cfg(config, hard=False)
         >>> print(config)
-        {'epochs': 50, 'lr0': 0.01, 'momentum': 1.2, 'save': False}  # corrected 'save' key
+        {'epochs': 50, 'lr0': 0.01, 'momentum': 0.937, 'save': False}  # corrected 'save' key
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/__init__.py]: 📝 **LOW**: The example output still looks inconsistent with `check_cfg(..., hard=False)`: if `save` is the string `"true"`, coercion should not produce `False`. As written, the docstring still misstates the function's behavior for boolean conversion.

**Suggested change:**
```suggestion
        {'epochs': 50, 'lr0': 0.01, 'momentum': 0.937, 'save': True}  # corrected 'save' key
```

File: docs/en/datasets/semantic/index.md (hunk 1 above)
Comment (glenn-jocher) [reply docs/en/datasets/semantic/index.md]: Good point — updated in 99b8b1791 to keep the fallback behavior factual while recommending lossless formats only (.png/.tiff), with a note that lossy compression corrupts class ID values.

File: examples/YOLOv8-CPP-Inference/README.md (hunk 3 above)
Comment (glenn-jocher) [reply examples/YOLOv8-CPP-Inference/README.md:56]: Fixed in 99b8b1791 — the command comment now says to run from the root of the cloned yolov5 repository, so the block is self-consistent.

File: ultralytics/cfg/__init__.py (hunk 4 above)
Comment (glenn-jocher) [reply ultralytics/cfg/__init__.py]: Applied in 99b8b1791. Confirmed in the code: the bool branch does cfg[k] = bool(v), and bool("true") is True, so the corrected output is {'save': True}.

File: docs/en/models/fast-sam.md (hunk 2 above)
Comment (glenn-jocher) [reply docs/en/models/fast-sam.md:195]: The single-class requirement is already called out directly below the examples: "FastSAM only supports detection and segmentation of a single class... you need to convert all object category IDs to 0." That note covers both tabs, and the Python tab has used coco8-seg.yaml here all along — this change only aligned the CLI tab with it (coco8.yaml has no masks at all, so it was strictly worse for segment val). There is no stock single-class seg dataset that would make a cleaner default here (crack-seg is single-class but requires a download), so keeping coco8-seg with the existing remapping note seems like the right trade-off. Happy to switch both tabs to crack-seg.yaml if you prefer.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Thanks @glenn-jocher for this wide-ranging documentation cleanup and accuracy pass across YOLO26, datasets, exports, solutions, and examples.

As Leonardo da Vinci said, *“Details make perfection, and perfection is not a detail.”* This PR is a great example of that mindset in action—small fixes to commands, arguments, paths, defaults, and explanations add up to a much clearer and more trustworthy experience for the whole Ultralytics community.

Really appreciate the care put into aligning the docs with current behavior and making the repo more accessible for both new and experienced users.
